deepmd/pt/model/descriptor/descriptor.py

```diff
@@ -332,7 +360,50 @@ def forward(
         mapping: Optional[torch.Tensor] = None,
     ):
         """Calculate DescriptorBlock."""
-        raise NotImplementedError
+        pass
+
+    # may have a better place for this method...
+    def build_type_exclude_mask(
+        self,
+        nlist: torch.Tensor,
+        atype_ext: torch.Tensor,
+    ) -> torch.Tensor:
```

Comment:
Maybe we can make `ExcludeMask` class in utils and as an attribute (if needed) in atomic model level? Then we can get `exlude_mask` at the begining of model `forward` and pass the mask to `forward` of both descriptor and fitting.

Reply:
Anyway, this can also be done in spin PR.

Reply:
A `ExcludeMask` class like that in the `dpmodel` cannot be jitted...
the exclude_mask is only needed by descriptors, because it is pairwise. The fitting needs only single atom masks.